File: lambda/y_combinator.js
const Y = f => (x => x(x))(x => f(y => x(x)(y)))
const fib_gen = f => n => n < 2 ? 1 : f(n - 1) + f(n - 2)
const power_gen = f => (n => n * n)
console.log(Y(power_gen)(10))
for (let index = 0; index < 10; index++) {
    console.log(Y(fib_gen)(index))
}

File: lambda/y_combinator.py
Y = lambda f:(lambda x:x(x))(lambda y:f(lambda *args:y(y)(*args)))
power_gen = lambda f:lambda n:n*n
print(Y(power_gen)(10))
fac_gen = lambda f:lambda n: 1 if n < 2 else f(n-1) + f(n-2)
print([Y(fac_gen)(i) for i in range(10)])
